wagtail/admin/views/pages/create.py

```diff
@@ -189,6 +190,7 @@ def save_action(self):
         # remain on edit page for further edits
         return self.redirect_and_remain()
 
+    @transaction.atomic
```

Comment:
OOI have you seen any commentary online why they don't make this atomic? Just wondering if there could be a downside

Reply:
There's a performance cost to making a wider range of DB operations part of the same transaction I believe. Some parts of wagtail do use transactions, but the scope seems to be limited. I have read that nested transactions are fine, so it shouldn't cause any issues: https://docs.djangoproject.com/en/5.0/topics/db/transactions/